alarm/Exception.js

```diff
@@ -185,28 +194,21 @@ module.exports = class {
             continue;
           }
         }
-        let valArray = val, val2Array = val2;
-        isJsonString(val) && (valArray = JSON.parse(val));
+        let val2Array = val2;
+        // Exception will be parsed at object creation
+        // while alarm will always use string to avoid compatibility issue with clients
         isJsonString(val2) && (val2Array = JSON.parse(val2));
-        if (key && key.startsWith("p.tag.ids")) {
-          const intersect = _.intersection(valArray, val2Array);
-          if (intersect.length > 0) {
+
+        if (key.startsWith("p.tag.ids")) {
+          if (_.intersection(val, val2Array).length > 0) {
```

Comment:
should convert to `[string]` if p.tag.ids doesn't convert when creating alarm instance.